src/openslide-vendor-dicom.c

```diff
@@ -711,6 +726,76 @@ static void add_properties(openslide_t *osr, struct dicom_level *level0) {
   }
 }
 
+static bool quickhash_level_pixels(struct _openslide_hash *hash,
+                                   struct dicom_level *level) {
+  double d;
+  if (!get_tag_decimal_str(level->file->metadata, NumberOfFrames, 0, &d)) {
+    return false;
+  }
+  uint32_t num_frames = d;
+
+  int64_t total_bytes = 0;
+  for (uint32_t i = 0; i < num_frames; i++) {
+    DcmFrame *frame = dcm_filehandle_read_frame(NULL,
+                                                level->file->filehandle,
+                                                i + 1);
+    if (frame == NULL) {
+      return false;
+    }
+
+    total_bytes += dcm_frame_get_length(frame);
+    if (total_bytes > (5 << 20)) {
+      return false;
+    }
+
+    _openslide_hash_data(hash,
+                         dcm_frame_get_value(frame),
+                         dcm_frame_get_length(frame));
+
+    dcm_frame_destroy(frame);
+  }
+
+  return true;
+}
+
+static void hash_int64(struct _openslide_hash *hash, int64_t i) {
+  // always hash in network byte order
+  int64_t i_be = GINT64_TO_BE(i);
+  _openslide_hash_data(hash, &i_be, sizeof(i_be));
+}
+
+// the required string-valued DICOM tags
+static const char *quickhash_level_metadata_tags[] = {
+  StudyDate,
+  StudyTime,
+  PatientID,
+  StudyID,
+  SeriesNumber,
+  ReferringPhysicianName,
+  PatientName,
+  PatientBirthDate,
+  PatientSex,
+  PatientOrientation,
+};
+
+static void quickhash_level_metadata(struct _openslide_hash *hash,
+                                     struct dicom_level *level) {
+  hash_int64(hash, level->base.w);
+  hash_int64(hash, level->base.h);
+  hash_int64(hash, level->base.tile_w);
+  hash_int64(hash, level->base.tile_h);
```

Comment:
Would it make sense to include the underlying attributes in `quickhash_level_metadata_tags` instead of special-casing the values here?  These struct fields reflect OpenSlide's presentation of the slide, and it's conceivable that in the future they won't always be equal to the underlying attributes.

Reply:
OK, I added a second table for int-valued tags.